app/jobs/appointment_reminders/send_reminder_job.rb

```diff
@@ -0,0 +1,98 @@
+class AppointmentReminders::SendReminderJob
+  include Rails.application.routes.url_helpers
+  include Sidekiq::Worker
+
+  sidekiq_options queue: :high
+
+  def perform(reminder_id)
+    reminder = AppointmentReminder.includes(:appointment, :patient).find(reminder_id)
+    if reminder.status != "scheduled"
+      report_error("scheduled appointment reminder has invalid status")
+    else
+      send_message(reminder)
+    end
+  end
+
+  private
+
+  def send_message(reminder)
+    notification_service = NotificationService.new
+
+    begin
+      response = if communication_type == "missed_visit_whatsapp_reminder"
+        notification_service.send_whatsapp(
+          phone_number(reminder.patient),
+          appointment_message(reminder),
+          callback_url
+        )
+      else
+        notification_service.send_sms(
+          phone_number(reminder.patient),
+          appointment_message(reminder),
+          callback_url
+        )
+      end
+
+      Communication.create_with_twilio_details!(
+        appointment: reminder.appointment,
+        appointment_reminder: reminder,
+        twilio_sid: response.sid,
+        twilio_msg_status: response.status,
+        communication_type: communication_type
+      )
+    rescue Twilio::REST::TwilioError => e
+      report_error(e)
+    end
+  end
+
+  def appointment_message(reminder)
+    I18n.t(
+      reminder.message,
+      assigned_facility_name: reminder.appointment.facility.name,
+      patient_name: reminder.patient.full_name,
+      appointment_date: reminder.appointment.scheduled_date,
+      locale: patient_locale(reminder.patient)
+    )
+  end
+
+  # i'm not convinced this is good enough
```

Comment:
The reason I'm not convinced this is good enough is that this is the logic in address:
```
  STATE_TO_LOCALE = {
    punjab: "pa-Guru-IN",
    maharashtra: "mr-IN",
    karnataka: "kn-IN"
  }

  def locale
    STATE_TO_LOCALE.fetch(state_to_sym, :en)
  end
```
That doesn't seem to support all options.